radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleySafetyStoreStore.java

```diff
@@ -0,0 +1,139 @@
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.store.berkeley;
+
+import com.google.inject.Inject;
+import com.radixdlt.consensus.Vote;
+import com.radixdlt.consensus.bft.View;
+import com.radixdlt.consensus.safety.PersistentSafetyStateStore;
+import com.radixdlt.consensus.safety.SafetyState;
+import com.radixdlt.utils.Longs;
+import com.radixdlt.utils.Pair;
+import com.sleepycat.je.Database;
+import com.sleepycat.je.DatabaseConfig;
+import com.sleepycat.je.DatabaseEntry;
+import com.sleepycat.je.Environment;
+import com.sleepycat.je.LockMode;
+import com.sleepycat.je.OperationStatus;
+import com.sleepycat.je.Transaction;
+
+import java.nio.ByteBuffer;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.radix.database.DatabaseEnvironment;
+
+import java.util.Objects;
+
+/**
+ * Store which persists state required to preserve the networks safety in case of a
+ * node restart.
+ *
+ * TODO: Prune saved safety state.
+ */
+public final class BerkeleySafetyStoreStore implements PersistentSafetyStateStore {
+	private static final String SAFETY_STORE_NAME = "safety_store";
+	private static final Logger logger = LogManager.getLogger();
+
+	private final DatabaseEnvironment dbEnv;
+	private Database safetyStore;
+
+	@Inject
+	public BerkeleySafetyStoreStore(DatabaseEnvironment dbEnv) {
+		this.dbEnv = Objects.requireNonNull(dbEnv, "dbEnv is required");
+
+		this.open();
+	}
+
+	private void fail(String message) {
+		logger.error(message);
+		throw new BerkeleyStoreException(message);
+	}
+
+	private void fail(String message, Exception cause) {
+		logger.error(message, cause);
+		throw new BerkeleyStoreException(message, cause);
+	}
+
+	private void open() {
+		DatabaseConfig primaryConfig = new DatabaseConfig();
+		primaryConfig.setAllowCreate(true);
+		primaryConfig.setTransactional(true);
+
+		try {
+			// This SuppressWarnings here is valid, as ownership of the underlying
+			// resource is not changed here, the resource is just accessed.
+			@SuppressWarnings("resource")
+			Environment env = this.dbEnv.getEnvironment();
+			this.safetyStore = env.openDatabase(null, SAFETY_STORE_NAME, primaryConfig);
+		} catch (Exception e) {
+			throw new BerkeleyStoreException("Error while opening database", e);
+		}
+
+		if (System.getProperty("db.check_integrity", "1").equals("1")) {
+			// TODO implement intergrity check
+		}
+	}
+
+	public Optional<Pair<Long, SafetyState>> get() {
+		try (com.sleepycat.je.Cursor cursor = this.safetyStore.openCursor(null, null)) {
+			DatabaseEntry pKey = new DatabaseEntry();
+			DatabaseEntry value = new DatabaseEntry();
+			OperationStatus status = cursor.getLast(pKey, value, LockMode.DEFAULT);
+			if (status == OperationStatus.SUCCESS) {
+				long lockedView = Longs.fromByteArray(value.getData());
+				long epochFound = ByteBuffer.wrap(pKey.getData()).getLong();

```

Comment:
Just use `Longs.fromByteArray(pkey.getData(), offset)`.  Cycling the array through a ByteBuffer just produces more garbage, particularly as it is done twice here.